server/apps/immich/src/api-v1/asset/dto/asset-check.dto.ts

```diff
@@ -0,0 +1,19 @@
+import { Type } from 'class-transformer';
+import { IsArray, IsNotEmpty, IsString, ValidateNested } from 'class-validator';
+
+export class AssetBulkUploadCheckItem {
+  @IsString()
+  @IsNotEmpty()
+  id!: string;
```

Comment:
What is the would we fill in for the ID for the incoming asset?

Reply:
The client is free to choose any id that makes sense for it

Reply:
Instead of assuming the return/response order is going to be the same, the client sends an ID to keep track of individual items. The ID is returned back along with the result (accepted/rejected).

Reply:
Ah got it. Thank you for clarifying!